### PR TITLE
Chakra setup

### PR DESCRIPTION
**Pull request by ozzydogandev**
-Chakra setup is completed

### CODE REVIEW

File: src/components/ui/close-button.jsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import { IconButton as ChakraIconButton } from '@chakra-ui/react'
+import * as React from 'react'
+import { LuX } from 'react-icons/lu'
+
+export const CloseButton = React.forwardRef(function CloseButton(props, ref) {
+  return (
+    <ChakraIconButton variant='ghost' aria-label='Close' ref={ref} {...props}>
+      {_nullishCoalesce(props.children, () => (
+        <LuX />
+      ))}
```
Comment (robstrader) [inline src/components/ui/close-button.jsx:17]: Kind of curious, what's up with the _nullishCoalesce function? I see what it is doing, I'm just making sure you know 😄 .

**Review by robstrader — COMMENTED** [review]
I don't think you need to include all these reusable components right now. I would add them as they are needed(you can always look back at this commit in this PR and copy them). How are you experimenting with these controls?

There is a funny acronym called YAGNI. It means Ya Ain't Gonna Need It. Try not to add anything until you are using it.

I like seeing this overall though. What do you think of Chakra UI?

**Comment by robstrader** [comment]
Also, I'm thinking about the directory structure. I think moving things around such that there is a `views` folder with the main views like `HomePage` and then reusable components would be in the `components` folder. What do you think? I don't particularly like naming a folder `ui`, it's all UI.

**Comment by ozzydogandev** [comment]
Hi Rob, thanks for checking my PR! I wanted to clarify that we need the reusable components in src/components/ui because we won't be able to fully utilize Chakra UI without them. The components, like Field, came from the `npx @chakra-ui/cli snippet add` command, which helps streamline how we use Chakra components.

For example:

```
import { Input } from "@chakra-ui/react";
**import { Field } from "@/components/ui/field";**

const Demo = () => {
  return (
    <Field label="Email" required helperText="We'll never share your email.">
      <Input placeholder="Enter your email" />
    </Field>
  );
};
```

Also, I agree with your suggestion to reorganize the folders. I've updated the directory structure as follows and already committed and pushed the changes 🫡 :

-Moved main pages like HomePage.jsx into a new src/views folder.
-Kept reusable components (navbar.jsx) in src/components.

And I hope I never include YAGNI codes in my work because it's something I always hear from my wife lol

**Review by robstrader — APPROVED** [review]
Interesting, I hadn't use the Chakra CLI before, but I do know CLIs were the new 'thing' that a few of the component libraries were doing. We might want to clean up the ones we don't need in the future.

**Pull request merged**